perception/multi_object_tracker/include/multi_object_tracker/utils/utils.hpp

```diff
@@ -295,40 +295,43 @@ inline void calcAnchorPointOffset(
  * @param input_object: input convex hull objects
  * @param output_object: output bounding box objects
  */
-inline void convertConvexHullToBoundingBox(
+inline bool convertConvexHullToBoundingBox(
   const autoware_auto_perception_msgs::msg::DetectedObject & input_object,
   autoware_auto_perception_msgs::msg::DetectedObject & output_object)
 {
-  const Eigen::Vector2d center{
-    input_object.kinematics.pose_with_covariance.pose.position.x,
-    input_object.kinematics.pose_with_covariance.pose.position.y};
-  const auto yaw = tf2::getYaw(input_object.kinematics.pose_with_covariance.pose.orientation);
-  const Eigen::Matrix2d R_inv = Eigen::Rotation2Dd(-yaw).toRotationMatrix();
+  // check footprint size
+  if (input_object.shape.footprint.points.size() < 3) {
+    return false;
+  }
 
+  // look for bounding box boundary
   double max_x = 0;
   double max_y = 0;
   double min_x = 0;
   double min_y = 0;
   double max_z = 0;
-
-  // look for bounding box boundary
   for (size_t i = 0; i < input_object.shape.footprint.points.size(); ++i) {
-    Eigen::Vector2d vertex{
-      input_object.shape.footprint.points.at(i).x, input_object.shape.footprint.points.at(i).y};
-
-    const Eigen::Vector2d local_vertex = R_inv * (vertex - center);
-    max_x = std::max(max_x, local_vertex.x());
-    max_y = std::max(max_y, local_vertex.y());
-    min_x = std::min(min_x, local_vertex.x());
-    min_y = std::min(min_y, local_vertex.y());
-
-    max_z = std::max(max_z, static_cast<double>(input_object.shape.footprint.points.at(i).z));
+    const double foot_x = input_object.shape.footprint.points.at(i).x;
```

Comment:
[imho]
I think footprint is calculated by relative distance from centroid in base_link and not in object coordinate.
So, I guess yaw rotation fix will be necessary before get maximum x,y value.

Currently, it won't make any change because yaw of unknown object may be 0 in all case.

Reply:
Base line: the source of the unknown polygon object is from `shape_estimator`.

1. Object Position: 
   * Coordinate origin: base_link
   * Position: polygon centroid 
   * [evidence](https://github.com/autowarefoundation/autoware.universe/blob/9dcf81ec79c7f465c547196148f8c9c40e6d3411/perception/shape_estimation/lib/model/convex_hull.cpp#L88-L89)
2. Footprint:
   * Coordinate origin: object position (polygon centroid)
   * Positions: convex hull point positions, local
   * [evidence](https://github.com/autowarefoundation/autoware.universe/blob/9dcf81ec79c7f465c547196148f8c9c40e6d3411/perception/shape_estimation/lib/model/convex_hull.cpp#L75-L81)

I also checked the object size, output of this function, and confirmed that the numbers are as expected (about 0.7 m).

Therefore the footprint points are not from the base_link, but from the object origin.

Reply:
If the message defines the footprint is in the object coordinate (as its position is from the object position), the width and length should be on its coordinate, not rotated one.

It doesn't matter whether the `shape_estimator` implementation is done in `yaw = 0` or not.
If the `shape_estimator` do the yaw estimation in the polygon&convex hull function, it should rotate the point cloud when it make the message.

Reply:
Understood. Then, your code is fine.